source/sixtrack_input.f90

```diff
@@ -420,15 +434,29 @@ subroutine sixin_parseInputLineSETT(inLine, iLine, iErr)
     case("text")
       st_finalstate = 2
     case default
-      write(lout,"(a)") "INPUT> ERROR FINALSTATE type must be either 'binary' or 'text', got '"//trim(lnSplit(2))//"'"
+      write(lout,"(a)") "INPUT> ERROR FINALSTATE first value must be either 'binary' or 'text', got '"//trim(lnSplit(2))//"'"
       iErr = .true.
       return
     end select
-    if(st_finalstate == 1) then
+    if(nSplit == 3) then
+      if(lnSplit(3) == "ions") then
+        st_finalstate = st_finalstate + 2
+      else
+        write(lout,"(a)") "INPUT> ERROR FINALSTATE second value must be 'ions', got '"//trim(lnSplit(3))//"'"
+        iErr = .true.
+        return
+      end if
+    end if
+    select case(st_finalstate)
+    case(1)
       write(lout,"(a,i0)") "INPUT> Particle final state will be dumped as a binary file"
-    else
+    case(2)
       write(lout,"(a,i0)") "INPUT> Particle final state will be dumped as a text file"
-    end if
+    case(3)
+      write(lout,"(a,i0)") "INPUT> Particle final state will be dumped as a binary file with ion data included"
+    case(4)
+      write(lout,"(a,i0)") "INPUT> Particle final state will be dumped as a text file with ion data included"
+    end select
```

Comment:
I am not a big fan of variables encoding multiple states of different logical flags... I would have had two boolean vars instead. But this is I guess rater a problem of taste...